### PR TITLE
Add multi-vari distance calculation

### PR DESCRIPTION
**Pull request by evanna00**
**PR Checklist**

- [x] This comment contains a description of changes (with reason)
- [x] Referenced issue is linked
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [x] Documentation in `docs` is updated
- [x] `docs/changelog.md` is updated

**Description of changes**

Calculate the multi-vary distance and specify the outliers.

**Related Issues**

 Issue https://github.com/ewels/MegaQC/issues/433

### CODE REVIEW

**Comment by multimeric** [comment]
We already had outlier detection previously. The change you made is more broad than that.

File: megaqc/rest_api/plot.py [hunk 1]
```diff
@@ -27,129 +29,90 @@ def trend_data(fields, filter, plot_prefix, control_limits, center_line):
...
+
+    cov = EmpiricalCovariance()
+    y = y.reshape(-1, len(fields))
+
+    cov.fit(y)
+    distance = cov.mahalanobis(y)
+
+    # Calculate the critical value
+    n, p = y.shape
+    cri = f.isf(0.05, dfn=p, dfd=n - p)
```
Comment (multimeric) [inline megaqc/rest_api/plot.py:73]: Add a comment here explaining what you are doing.

File: megaqc/rest_api/plot.py [hunk 2]
```diff
@@ -27,129 +29,90 @@ def trend_data(fields, filter, plot_prefix, control_limits, center_line):
...
+        dict(
+            id="inliers",
+            type="scatter",
+            text=names,
+            hoverinfo="text+x+y",
+            x=x[distance < line],
+            y=distance[distance < line],
+            line=dict(color="rgb(0,0,250)"),
+            mode="markers",
+            name="Inliers".format(data_type),
```
Comment (multimeric) [inline megaqc/rest_api/plot.py:87]: You no longer need to use `.format` if you aren't interpolating any values.

File: megaqc/rest_api/plot.py [hunk 3]
```diff
@@ -1,9 +1,11 @@
+import math
```
Comment (multimeric) [inline megaqc/rest_api/plot.py:1]: Check your imports to ensure you are not importing modules that you don't use.

**Review by multimeric — CHANGES_REQUESTED** [review]

File: megaqc/rest_api/plot.py (hunk 1 above)
Comment (evanna00) [reply megaqc/rest_api/plot.py:72]: Calculate the line of critical value for the distribution.

Comment (multimeric) [reply megaqc/rest_api/plot.py:72]: Yes, but some more detail would be good explain the statistics being used here. You might want to refer to: https://online.stat.psu.edu/stat505/lesson/7/7.1/7.1.3

Comment (evanna00) [reply megaqc/rest_api/plot.py:72]: Using Hotelling’s T-Square to test the statistic for the F distribution.

**Pull request merged**